src/backend/distributed/commands/foreign_constraint.c

```diff
@@ -141,39 +148,56 @@ ErrorIfUnsupportedForeignConstraintExists(Relation relation, char referencingDis
 
 		/* set referenced table related variables here if table is referencing itself */
 
+		char referencedReplicationModel = REPLICATION_MODEL_INVALID;
 		if (!selfReferencingTable)
 		{
 			referencedDistMethod = PartitionMethod(referencedTableId);
 			referencedDistKey = (referencedDistMethod == DISTRIBUTE_BY_NONE) ?
 								NULL :
 								DistPartitionKey(referencedTableId);
 			referencedColocationId = TableColocationId(referencedTableId);
+			referencedReplicationModel = TableReplicationModel(referencedTableId);
 		}
 		else
 		{
 			referencedDistMethod = referencingDistMethod;
 			referencedDistKey = referencingDistKey;
 			referencedColocationId = referencingColocationId;
+			referencedReplicationModel = referencingReplicationModel;
 		}
 
-		bool referencingIsReferenceTable = (referencingDistMethod == DISTRIBUTE_BY_NONE);
-		bool referencedIsReferenceTable = (referencedDistMethod == DISTRIBUTE_BY_NONE);
+		bool referencingIsNoneDistTable = (referencingDistMethod == DISTRIBUTE_BY_NONE);
```

Comment:
in that context, we really use that variable to check if the relation is distribute by none, a few lines below, we do such checks by using some other variables

Reply:
Looking at the error messages etc, it seems like what we care is understanding if the table is citus local or reference table. Could you elaborate where we care about DISTRIBUTE_BY_NONE?

```c
		if (referencingIsNoneDistTable && !referencedIsNoneDistTable)
		{
			ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
							errmsg("cannot create foreign key constraint "
								   "since foreign keys from reference tables "
								   "to distributed tables are not supported"),
							errdetail("A reference table can only have foreign "
									  "keys to other reference tables and citus "
									  "local tables")));
		}
```

Reply:
For example, in the snipped you shared, we error out if the referenced table is a `NoneDistTable`, that means, it is a distributed table and we error out for such foreign keys to distributed tables.

Reply:
Discussed via chat, decided to rename the variable to `referencingIsCitusLocalOrRefTable` etc.. Open to better ideas (if any)